File: shopbill-services/src/main/java/com/shop/bill/entity/Product.java
package com.shop.bill.entity;

import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Getter;
import lombok.Setter;

/**
*
* @author Akhil Singla
*/
@Getter
@Setter
@Entity
public class Product extends CommonEntity {
    private String code;
    private String name;
    private String hsnCode;
    private Double unit;
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "unit_type")
	private CommonCode unitType;
    private Double price;
}

File: shopbill-ui/src/app/shared/models/Product.ts
import { Common } from './Common'
import { CommonCode } from './CommonCode'

export class Product extends Common {
    code: string;
    name: string;
    hsnCode: string;
    unit: number;
    unitType: CommonCode;
    price: number;

    constructor(id: number, delYn: string, crtId: string, uptId: string, crtDtm: number, uptDtm: number, 
        code: string, name: string, hsnCode: string, unit: number, unitType: CommonCode, price: number) {
        super(id, delYn, crtId, uptId, crtDtm, uptDtm);
        this.code = code;
        this.name = name;
        this.hsnCode = hsnCode;
        this.unit = unit;
        this.unitType = unitType;
        this.price = price;
    }
}